# Farasa diacritization error

## Undiacritize the dataset

In [2]:
with open("text_cla.txt") as f:
    text_cla = f.read().splitlines()
with open("text_asc.txt") as f:
    text_asc = f.read().splitlines()
with open("text_mdp.txt") as f:
    text_mdp = f.read().splitlines()

In [11]:
%cd catt
from utils import strip_tashkeel

undiac_cla = [strip_tashkeel(i) for i in text_cla]
undiac_asc = [strip_tashkeel(i) for i in text_asc]
undiac_mdp = [strip_tashkeel(i) for i in text_mdp]
%cd ..

[Errno 2] No such file or directory: 'catt'
/home/auriza.akbar/nlp806-proj/catt
/home/auriza.akbar/nlp806-proj


In [12]:
with open("undiac_cla.txt", "w") as f:
    for x in undiac_cla: f.write(f"{x}\n")
with open("undiac_asc.txt", "w") as f:
    for x in undiac_asc: f.write(f"{x}\n")
with open("undiac_mdp.txt", "w") as f:
    for x in undiac_mdp: f.write(f"{x}\n")

## Diacritize with Farasa

In [16]:
%%bash
java -jar farasa/FarasaDiacritizeJar.jar -i undiac_cla.txt -o farasa_cla.txt
java -jar farasa/FarasaDiacritizeJar.jar -i undiac_asc.txt -o farasa_asc.txt
java -jar farasa/FarasaDiacritizeJar.jar -i undiac_mdp.txt -o farasa_mdp.txt

Initializing the system ....
System ready!               
Initializing the system ....
System ready!               
Initializing the system ....
System ready!               


## Calculate DER

In [18]:
%%bash
echo ClArTTS
python catt/compute_der.py text_cla.txt farasa_cla.txt no
python catt/compute_der.py text_cla.txt farasa_cla.txt yes
echo ASC
python catt/compute_der.py text_asc.txt farasa_asc.txt no
python catt/compute_der.py text_asc.txt farasa_asc.txt yes
echo MDPC
python catt/compute_der.py text_mdp.txt farasa_mdp.txt no
python catt/compute_der.py text_mdp.txt farasa_mdp.txt yes

ClArTTS
Total DER: %14.502
Total WER: %58.116
Total mismatch samples: 29
-----------------------------------------------------------------------------------------
Total DER: %16.550
Total WER: %62.850
Total mismatch samples: 29
-----------------------------------------------------------------------------------------
ASC
Total DER: %18.744
Total WER: %67.583
Total mismatch samples: 1
-----------------------------------------------------------------------------------------
Total DER: %21.400
Total WER: %72.480
Total mismatch samples: 1
-----------------------------------------------------------------------------------------
MDPC
Total DER: %32.562
Total WER: %93.131
Total mismatch samples: 40
-----------------------------------------------------------------------------------------
Total DER: %37.864
Total WER: %98.066
Total mismatch samples: 40
-----------------------------------------------------------------------------------------


## Results

| Dataset | DER-NoCE | WER-NoCE | DER-CE | WER-CE |
|---------|----------|----------|--------|--------|
| ClArTTS | 14.50    | 58.12    | 16.55  | 62.85  |
| ASC     | 18.74    | 67.58    | 21.40  | 72.48  |
| MDPC    | 32.56    | 93.13    | 37.86  | 98.07  |